pkg/cmd/ssh/command.go

```diff
@@ -0,0 +1,541 @@
+	cmd.Flags().StringVar(&o.SSHPublicKeyFile, "public-key-file", "", "Path to the file that contains a public SSH key. If not given, a temporary keypair will be generated.")
+	cmd.Flags().DurationVar(&o.WaitTimeout, "wait-timeout", o.WaitTimeout, "Maximum duration to wait for the bastion to become available.")
+	cmd.Flags().BoolVar(&o.KeepBastion, "keep-bastion", o.KeepBastion, "Do not delete immediately when gardenctl exits (Bastions will be garbage-collected after some time)")
+
+	return cmd
+}
+
+func runCommand(f util.Factory, o *Options) error {
+	manager, err := f.Manager()
+	if err != nil {
+		return err
+	}
+
+	// validate the current target
+	currentTarget, err := manager.CurrentTarget()
+	if err != nil {
+		return err
+	}
+
+	if currentTarget.ShootName() == "" {
+		return errors.New("no Shoot cluster targeted")
+	}
+
+	// create client for the garden cluster
+	gardenClient, err := manager.GardenClient(currentTarget)
+	if err != nil {
+		return err
+	}
+
+	// fetch targeted shoot (ctx is cancellable to stop the keep alive goroutine later)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	shoot, err := util.ShootForTarget(ctx, gardenClient, currentTarget)
+	if err != nil {
+		return err
+	}
+
+	// fetch the SSH key for the shoot nodes
+	nodePrivateKey, err := getShootNodePrivateKey(ctx, gardenClient, shoot)
+	if err != nil {
+		return err
+	}
+
+	// save the key into a temporary file that we try to clean up when exiting
+	nodePrivateKeyFile, err := savePublicKey(nodePrivateKey)
+	if err != nil {
+		return err
+	}
+
+	defer os.Remove(nodePrivateKeyFile)
+
+	// if a node was given, check if the node exists
+	// and if not, exit early and do not create a bastion
+	node, err := getShootNode(ctx, o, manager, currentTarget)
+	if err != nil {
+		return err
+	}
+
+	// prepare Bastion resource
+	policies := []operationsv1alpha1.BastionIngressPolicy{}
+
+	for _, cidr := range o.CIDRs {
+		policies = append(policies, operationsv1alpha1.BastionIngressPolicy{
+			IPBlock: networkingv1.IPBlock{
+				CIDR: cidr,
+			},
+		})
+	}
+
+	sshPublicKey, err := ioutil.ReadFile(o.SSHPublicKeyFile)
+	if err != nil {
+		return fmt.Errorf("failed to read SSH public key: %v", err)
+	}
+
+	// avoid GenerateName because we want to immediately fetch and check the bastion
+	bastionID, err := utils.GenerateRandomString(8)
+	if err != nil {
+		return fmt.Errorf("failed to create bastion name: %v", err)
+	}
+
+	bastionName := fmt.Sprintf("cli-%s", strings.ToLower(bastionID))
+	bastion := &operationsv1alpha1.Bastion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bastionName,
+			Namespace: shoot.Namespace,
+		},
+		Spec: operationsv1alpha1.BastionSpec{
+			ShootRef: corev1.LocalObjectReference{
+				Name: shoot.Name,
+			},
+			SSHPublicKey: strings.TrimSpace(string(sshPublicKey)),
+			Ingress:      policies,
+		},
+	}
+
+	fmt.Fprintf(o.IOStreams.Out, "Creating bastion %s…", bastion.Name)
```

Comment:
```bash
gardenctl ssh shoot--myproject--mynode
Auto-detected your system's CIDR as <redacted>/64
Creating bastion cli-uhgvgvnb…Waiting up to 10m0s for bastion to be ready…
Timed out waiting for the bastion to be ready.
Error: precondition failed
```
```bash
gardenctl-v2 ssh
Auto-detected your system's CIDR as <redacted>/64
Creating bastion cli-kpvvyqof…Waiting up to 10m0s for bastion to be ready…
An error occurred while waiting: bastions.operations.gardener.cloud "cli-kpvvyqof" not foundError: precondition failed
```
See the missing linebreaks above. And why do you use `fmt.Fprintln(o.IOStreams.Out, "")` instead of `\n` at the end of the previous `Fprintf` message?

Reply:
Because I like it when a status messag like "creating something..." keeps the cursor on the same line. But it's not important, I changed it now.